compilers/concrete-compiler/compiler/lib/Dialect/FHELinalg/IR/FHELinalgOps.cpp

```diff
@@ -1592,6 +1592,67 @@ mlir::LogicalResult FancyAssignOp::verify() {
   return mlir::success();
 }
 
+mlir::LogicalResult BroadcastOp::verify() {
+  auto inputType =
+      this->getInput().getType().dyn_cast_or_null<mlir::RankedTensorType>();
+  auto outputType =
+      this->getOutput().getType().dyn_cast_or_null<mlir::RankedTensorType>();
+
+  auto inputElementType = inputType.getElementType();
+  auto outputElementType = outputType.getElementType();
+
+  if (inputElementType != outputElementType) {
+    this->emitOpError() << "input element type " << inputElementType
+                        << " doesn't match output element type "
+                        << outputElementType;
+    return mlir::failure();
+  }
+
+  auto inputShape = inputType.getShape();
+  auto outputShape = outputType.getShape();
+
+  auto inputDimensions = inputShape.size();
+  auto outputDimensions = outputShape.size();
+
+  auto cannot_be_done = false;
+  if (inputDimensions > outputDimensions) {
+    cannot_be_done = true;
+    return mlir::failure();
+  } else {
+    auto modifiedInputShape = llvm::SmallVector<int64_t>();
+    for (size_t i = 0; i < outputDimensions - inputDimensions; i++) {
```

Comment:
This can be expressed more concisely through the constructor of `llvm::SmallVector`.

Reply:
What do you have in mind?

Reply:
What do you have in mind?

Reply:
If I am not mistaken, this adds `outputDimensions - inputDimensions` times `1` to `modifiedInputShape`. Changing line 1622 to `llvm::SmallVector<int64_t> modifiedInputShape(outputDimensions - inputDimensions, 1);` should have the same effect.

Reply:
Sure.